com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/DLSSPass.cs

```diff
@@ -333,18 +339,20 @@ private bool ShouldUseOptimalSettings()
 
             public void UpdateViewState(
                 in DlssViewData viewData,
+                /*uint backbufferWidth, uint backbufferHeight,*/
                 CommandBuffer cmdBuffer)
             {
-                bool shouldUseOptimalSettings = ShouldUseOptimalSettings();
+                bool shouldUseOptimalSettings = ShouldUseAutomaticSettings();
                 bool isNew = false;
                 if (viewData.outputRes != m_Data.outputRes ||
-                    (viewData.inputRes != m_Data.inputRes && !m_UsingOptimalSettings) ||
-                    (viewData.CanFitInput(m_Data.inputRes) && m_UsingOptimalSettings) ||
+                    (viewData.inputRes.width > m_BackbufferRes.width || viewData.inputRes.height > m_BackbufferRes.height) ||
+                    (viewData.inputRes != m_BackbufferRes && !m_OptimalSettingsRequest.CanFit(viewData.inputRes)) ||
```

Comment:
Is it fine to do this checks (m_OptimalSettingsRequest.CanFit) when optimal settings are not used?